src/ast/ast_validations.c

```diff
@@ -782,8 +782,21 @@ static AST_Validation _Validate_DELETE_Clauses(const AST *ast, char **reason) {
 	if(!delete_clause) return AST_VALID;
 
 	const cypher_astnode_t *match_clause = AST_GetClause(ast, CYPHER_AST_MATCH);
-	if(!match_clause) return AST_INVALID;
-
+	// Check for delete a projected entity such in "MATCH (n) WITH n AS x DELETE x".
+	const cypher_astnode_t *with_clause = AST_GetClause(ast, CYPHER_AST_WITH);
+	assert(!(with_clause && match_clause));
+	if(!match_clause && !with_clause) return AST_INVALID;
```

Comment:
Unfortunately I believe the following is valid:
`CREATE (a) DELETE a`
I would omit this check.